lib-utilities/common/workerpool.go

```diff
@@ -61,16 +61,14 @@ func RunWriteWorkers(jobChannel chan<- interface{}, dataBatch []interface{}, wor
 					done <- true
 					return
 				}
-				store.lock.Unlock()
 				data := store.getNextData()
 				jobChannel <- data
+        store.lock.Unlock()
 			}
 		}()
 	}
 }
 func (ds *dataBatchStore) getNextData() interface{} {
-	ds.lock.Lock()
```

Comment:
Why is this being removed?

Reply:
the getNextData() method has called inside the mutex lock. so we can't use lock inside another lock.